src/arrayMethodSort.js

```diff
@@ -4,9 +4,33 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let swapped;
+    let n = this.length;
+
+    do {
+      swapped = false;
+
+      for (let i = 0; i < n - 1; i++) {
+        if (compareFunction) {
+          if (compareFunction(this[i], this[i + 1]) > 0) {
+            [this[i], this[i + 1]] = [this[i + 1], this[i]];
+            swapped = true;
+          }
+        } else {
+          if (String(this[i]) > String(this[i + 1])) {
+            [this[i], this[i + 1]] = [this[i + 1], this[i]];
+            swapped = true;
+          }
+        }
+      }
+      n--;
+    } while (swapped);
+
+    return this;
   };
+
+  return this;
```

Comment:
The return statement here is unnecessary as `this` refers to the global object in this context, which isn't being modified or used after this function. You can safely remove this line.